tests/query_runner/test_google_spreadsheets.py

```diff
@@ -104,6 +109,40 @@ def test_parse_query(self):
         parsed = parse_query("key|0")
         self.assertEqual(("key", 0), parsed)
 
+    def test_parse_query_ignored(self):
+        parsed = parse_query("key")
+        self.assertEqual(("key", 0), parsed)
+
+        parsed = parse_query("key|")
+        self.assertEqual(("key", 0), parsed)
+
+        parsed = parse_query("key|1|")
+        self.assertEqual(("key", 0), parsed)
+
+    def test_parse_query_title(self):
+        parsed = parse_query('key|""')
+        self.assertEqual(("key", ""), parsed)
+
+        parsed = parse_query('key|"1"')
+        self.assertEqual(("key", "1"), parsed)
+
+        parsed = parse_query('key|"abc"')
+        self.assertEqual(("key", "abc"), parsed)
+
+        parsed = parse_query('key|"あ"')
+        self.assertEqual(("key", "あ"), parsed)
+
+        parsed = parse_query('key|"1""')
+        self.assertEqual(("key", '1"'), parsed)
+
+        parsed = parse_query('key|""')
+        self.assertEqual(("key", ""), parsed)
+
+    def test_parse_query_failed(self):
+        self.assertRaises(ValueError, parse_query, "key|0x01")
+        self.assertRaises(ValueError, parse_query, "key|a")
```

Comment:
I think a fair expectation would be that a single-word / no-word-boundaries string would also be interpreted as a worksheet name, but I think that's out of scope for now.